plugins/inventory/lxd.py

```diff
@@ -0,0 +1,1090 @@
+        urls = (url for url in url_list if self.validate_url(url))
+        for url in urls:
+            try:
+                socket_connection = LXDClient(url, self.client_key, self.client_cert, self.debug)
+                return socket_connection
+            except LXDClientException as err:
+                error_storage[url] = err
+        raise AnsibleError('No connection to the socket: {0}'.format(to_native(error_storage)))
+
+    def _get_networks(self):
+        """Get Networknames
+
+        Returns all network config names
+
+        Args:
+            None
+        Kwargs:
+            None
+        Raises:
+            None
+        Returns:
+            list(names): names of all network_configs"""
+        network_configs = self.socket.do('GET', '/1.0/networks')
+        # e.g. {'type': 'sync',
+        #       'status': 'Success',
+        #       'status_code': 200,
+        #       'operation': '',
+        #       'error_code': 0,
+        #       'error': '',
+        #       'metadata': ['/1.0/networks/lxdbr0']}
+        names = []
+        for index in network_configs['metadata']:
+            # e.g. ['', '1.0', 'networks', 'lxdbr0']
+            names.append(index.split('/')[3])
+        return names
+
+    def _get_containers(self):
+        """Get Containernames
+
+        Returns all containernames
+
+        Args:
+            None
+        Kwargs:
+            None
+        Raises:
+            None
+        Returns:
+            list(names): names of all containers"""
+        # e.g. {'type': 'sync',
+        #       'status': 'Success',
+        #       'status_code': 200,
+        #       'operation': '',
+        #       'error_code': 0,
+        #       'error': '',
+        #       'metadata': ['/1.0/containers/udemy-ansible-ubuntu-2004']}
+        containers = self.socket.do('GET', '/1.0/containers')
+        names = []
+        for index in containers['metadata']:
+            # e.g. ['', '1.0', 'containers', 'udemy-ansible-ubuntu-2004']
+            names.append(index.split('/')[3])
+        return names
+
+    def _get_config(self, branch, name):
+        """Get inventory of container
+
+        Get config of container
+
+        Args:
+            str(branch): Name oft the API-Branch
+            str(name): Name of Container
+        Kwargs:
+            None
+        Source:
+            https://github.com/lxc/lxd/blob/master/doc/rest-api.md
+        Raises:
+            None
+        Returns:
+            dict(config): Config of the container"""
+        config = {}
+        if isinstance(branch, tuple):
+            config[name] = {branch[1]: self.socket.do('GET', '/1.0/{0}/{1}/{2}'.format(to_native(branch[0]), to_native(name), to_native(branch[1])))}
+        else:
+            config[name] = {branch: self.socket.do('GET', '/1.0/{0}/{1}'.format(to_native(branch), to_native(name)))}
+        return config
+
+    def _merge_dicts(self, source, destination):
```

Comment:
Take a look at `from ansible.module_utils.common.dict_transformations import dict_merge` and surroundings, it is possible your flavour of merge is there already.

Reply:
Thank You